### PR TITLE
Settings menu

### PR DESCRIPTION
**Pull request by richjoregonstate**

### CODE REVIEW

File: ModelViewer/ModelViewerTest/test.cpp [hunk 1]
```diff
@@ -90,7 +90,7 @@ void ModelViewerTest::ResetViewAndShow()
 }
 
 
-void ModelViewerTest::integration()
+void ModelViewerTest::integration()// README: The addition of my settings menu borked this test
```
Comment (tigerman9854) [inline ModelViewer/ModelViewerTest/test.cpp]: I assume it's just the ordering of menu items.  Add another Key_Down and it should be fine :)

File: ModelViewer/ModelViewer/SettingsMenu.cpp [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+	panYSensitivity->setValidator(new QDoubleValidator(0.00001, 10000.0, 5));
+	xRotateSensitivity->setValidator(new QDoubleValidator(0.00001, 10000.0, 5));
+	yRotateSensitivity->setValidator(new QDoubleValidator(0.00001, 10000.0, 5));
+	zoomSensitivity->setValidator(new QDoubleValidator(0.00001, 10000.0, 5));
+	fieldOfView->setValidator(new QDoubleValidator(0.00001, 10000.0, 5));
+	nearPlane->setValidator(new QDoubleValidator(0.00001, 10000.0, 5));
+	farPlane->setValidator(new QDoubleValidator(0.00001, 10000.0, 5));
+	
+	// Add the fields to the layout
+	layout->addRow(tr("Pan X Sensitivty"), panXSensitivity);
```
Comment (tigerman9854) [inline ModelViewer/ModelViewer/SettingsMenu.cpp]: sensitivty -> sensitivity

File: ModelViewer/ModelViewer/SettingsMenu.cpp [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+	layout->addRow(tr("Pan Y Sensitivty"), panYSensitivity);
+	layout->addRow(tr("Rotate X Sensitivty"), xRotateSensitivity);
+	layout->addRow(tr("Rotate X Sensitivty"), yRotateSensitivity);
+	layout->addRow(tr("Zoom Sensitivty"), zoomSensitivity);
+	layout->addRow(tr("Feild of View"), fieldOfView);
```
Comment (tigerman9854) [inline ModelViewer/ModelViewer/SettingsMenu.cpp]: Feild -> Field

File: ModelViewer/ModelViewer/SettingsMenu.cpp [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+	: QWidget(parent)
+{
+	setWindowTitle(QString::fromLatin1("Settings"));
+	setContentsMargins(0, 0, 0, 0);
+	resize(400, 300);
+
+	m_pMainLayout = new QGridLayout(this);
+
+	// Set up the settings
+	settings = new QSettings("TBD", "Model Viewer");
```
Comment (tigerman9854) [inline ModelViewer/ModelViewer/SettingsMenu.cpp]: I think QSettings is static, so you don't have to pass around a pointer to the settings object.
You can just write `QSettings settings;` wherever you need access to the settings.  You still need to call `QSettings settings("TBD", "Model Viewer")` sometime at startup to initialize the app name: https://doc.qt.io/qt-5/qsettings.html#QSettings-4

**Review by tigerman9854 — COMMENTED** [review]
(Yes I know it's still WIP) Looking very good! Just a few small things I noticed.

**Comment by richjoregonstate** [comment]
Found it....
image

**Comment by richjoregonstate** [comment]
image

File: ModelViewer/ModelViewer/KeyBindEdit.cpp [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    else {
+        keySequence = defalutKeys;
+    }
+
+    setText(getSequence());
+}
+
+QString KeyBindEdit::getSequence()
+{
+    QString final = "";
```
Comment (tigerman9854) [inline ModelViewer/ModelViewer/KeyBindEdit.cpp]: `final` is a keyword in C++, recommend not using it as a variable name.

**Review by tigerman9854 — APPROVED** [review]
Looks good to me, probably just rename the var "final".

Feels good to use as well!

**Pull request merged**